frontend/packages/ceph-storage-plugin/README.md

```diff
@@ -0,0 +1,89 @@
+# OCS UI Features
+
+The OCS UI requires some annotations in the OCS Operator CSV to perform various actions.
+
+Following table maps the annotation to its use case and accepted values:
+|Annotation Name|Purpose|Accepted Values |
+|---------------------------------------|---------------------------|--------|
+| `features.ocs.openshift.io/enabled`| Activate UI Features | "external", "snapshot" |
+| `external.features.ocs.openshift.io/validation`| Mininum required keys to be supplied by the admin to connect to an external cluster | Array of Keys that need to be validated in UI |
+||||
+
+## Enabling Features in UI
+
+UI features are activated based on the values in `features.ocs.openshift.io/enabled` annotation. The following table maps a feature and the respective annotation required to activate it.
+| Feature |Feature guard|
+|------------------------------|------------|
+| External Cluster Installation| `external` |
+| Volume Snapshots| `snapshot`
+
+#### Example
+
+       "features.ocs.openshift.io/enabled": `["external"]`
+
+## JSON validation (Independent Mode)
+
+This is used to perform validation of the JSON file uploaded by the user during Independent mode installation. The values in `external.ocs.openshift.io/validation` annotation needs to be a stringified JSON. The JSON object should have the following keys and values:
+| Key | Accepted Values | Validation Performed |
+|------|-----------|-------------|
+| `secrets` | Array of String| JSON should contain all the `secrets` defined in the array. The `data.userKey` /`data.adminKey` field should be Base64 encoded and non-empty.
+|`configMaps`| Array of String| JSON should contain all the `configMaps` defined in the array. `data` field should be non-empty.
+| `storageClasses`| Array of String| JSON should contain all the `storageClasses` defined in the array. `data` field should be non-empty.
+
+#### Example
+
+    [
+
+    {
+
+        "kind": "ConfigMap",
+
+        "data": {
+
+    	    "maxMonId": "0",
+
+    	    "data": "a=10.106.31.93:6789",
+
+    	    "mapping": {}
+
+        },
+
+        "name": "rook-ceph-mon-endpoints"
+
+    },
+
+    {
+
+        "kind": "Secret",
+
+        "data": {
+
+    	    "mon-secret": "mon-secret",
+
+    	    "fsid": "ac1235ba-c390-48d7-8e1f-f2ece695b15b",
+
+    	    "cluster-name": "openshift-storage",
+
+    	    "admin-secret": "admin-secret"
```

Comment:
I don't see the property `data.userKey` /`data.adminKey` here in this example which is mandatory as per the validation.

Reply:
Validation for Base64 is not performed for all keys.

Reply:
That's fine. Since you have mentioned `The data.userKey /data.adminKey field should be Base64 encoded and non-empty.`, I don't see these fields here in the secret object.